### DISCUSSION TITLE
Use CUDA for Stable Diffusion inpainting?

### DISCUSSION THREAD

**Discussion by carco5a** [Sanster/IOPaint#452] [Q&A] [answered]
`iopaint start --model-dir D:\AI\IOPaint\model --model sd-v1-5-inpainting.ckpt --device=cuda --enable-interactive-seg --interactive-seg-model=vit_l  --interactive-seg-device=cuda --config D:\AI\IOPaint\config.json`

Interactive Seg and lama appear to use CUDA just fine. But SD inpaint max's the CPU and takes a while to complete even at 10 steps, which makes me think it's only using the CPU. Is it possible to use SD inpaint with CUDA?

**Comment by Sanster** [comment, marked as answer]
SD supports CUDA, you can check if the VRAM is occupied after the service starts.

**Reply by carco5a** [reply]
Thank you for this great tool and for your response, this was my mistake... I was used to A1111 inpaint speed at the same steps for a similar SD1.5 inpaint. I confirmed VRAM and GPU usage for  CUDA and when I forced IOPaint to use the CPU for SD inpaint on the same work it took **FAR** longer.
Thanks again.
